File: tools/smoke.py
# 浏览器路径 / 端口 / 依赖统一交给 tools/_browser.py 自动探测，不再写死
from _browser import URL, launch, need_playwright, check_server, reset_storage, max_steps

need_playwright()   # 缺依赖时给出可操作的提示，而不是 ImportError 堆栈
check_server()      # 确认静态服务器已经起来（启动服务器是使用者的动作）

from playwright.sync_api import sync_playwright  # noqa: E402 —— 故意放在依赖检查之后

errors = []

with sync_playwright() as p:
    b = launch(p)
    page = b.new_page(viewport={"width": 1280, "height": 860})
    page.on("console", lambda m: errors.append("CONSOLE: " + m.text) if m.type == "error" else None)
    page.on("pageerror", lambda e: errors.append("PAGEERROR: " + str(e)))

    reset_storage(page)
    page.wait_for_timeout(1200)

    def txt(sel):
        try:
            return page.text_content(sel)
        except Exception:
            return None

    print("首页时辰:", txt(".act-hour"))
    print("地点标题:", txt(".act-title"))
    print("当前视角:", txt(".view-who"))
    print("选项数量:", page.locator(".choice-btn").count())
    page.screenshot(path="tools/shot1_start.png")

    # 走一步
    page.locator(".choice-btn").first.click()
    page.wait_for_timeout(600)
    print("第二步时辰:", txt(".act-hour"))
    print("线索数:", page.locator("#clueList .list-item").count())
    page.screenshot(path="tools/shot2_step.png")

    # 自动点击到结局
    for _ in range(max_steps(page)):
        try:
            if not page.locator("#overlay").evaluate("el => el.classList.contains('hidden')"):
                break
        except Exception:
            pass
        btn = page.locator(".choice-btn").first
        if btn.count() == 0:
            break
        btn.click()
        page.wait_for_timeout(220)

    page.wait_for_timeout(600)
    try:
        end_visible = not page.locator("#overlay").evaluate("el => el.classList.contains('hidden')")
    except Exception:
        end_visible = False
    print("结局浮层:", end_visible)
    if end_visible:
        try:
            print("结局标签:", page.text_content(".end-tag"))
        except Exception:
            pass
    page.screenshot(path="tools/shot3_ending.png")
    b.close()

print("")
if errors:
    print("运行时错误:")
    for e in errors[:10]:
        print("   " + e)
else:
    print("OK: 无运行时错误")

File: tools/smoke.mjs
import { chromium } from 'playwright';

const url = 'http://127.0.0.1:8010/index.html';
const browser = await chromium.launch({ executablePath: 'C:/Program Files/Google/Chrome/Application/chrome.exe' });
const page = await browser.newPage({ viewport: { width: 1280, height: 860 } });

const errors = [];
page.on('console', m => { if (m.type() === 'error') errors.push('CONSOLE: ' + m.text()); });
page.on('pageerror', e => errors.push('PAGEERROR: ' + e.message));

await page.goto(url, { waitUntil: 'networkidle' });
await page.waitForTimeout(1200);

// 首页检查
const title = await page.textContent('.act-title').catch(() => null);
const hour = await page.textContent('.act-hour').catch(() => null);
const choiceCount = await page.locator('.choice-btn').count();
const name = await page.textContent('.view-who').catch(() => null);
console.log('首页时辰:', hour);
console.log('地点标题:', title);
console.log('当前视角:', name);
console.log('选项数量:', choiceCount);

await page.screenshot({ path: 'tools/shot1_start.png' });

// 点击第一个选项，走一步
await page.locator('.choice-btn').first().click();
await page.waitForTimeout(600);
const t2 = await page.textContent('.act-hour').catch(() => null);
const clueCount = await page.locator('#clueList .list-item').count();
console.log('第二步时辰:', t2);
console.log('线索数:', clueCount);
await page.screenshot({ path: 'tools/shot2_step.png' });

// 连续点击走到结局
for (let i = 0; i < 30; i++) {
  const overlayVisible = await page.locator('#overlay').evaluate(el => !el.classList.contains('hidden')).catch(() => false);
  if (overlayVisible) break;
  const btn = page.locator('.choice-btn').first();
  if (await btn.count() === 0) break;
  await btn.click();
  await page.waitForTimeout(220);
}
await page.waitForTimeout(600);
const endVisible = await page.locator('#overlay').evaluate(el => !el.classList.contains('hidden')).catch(() => false);
console.log('结局浮层显示:', endVisible);
if (endVisible) {
  const endTitle = await page.textContent('.end-tag').catch(() => null);
  console.log('结局:', endTitle);
}
await page.screenshot({ path: 'tools/shot3_ending.png' });

console.log('');
if (errors.length) {
  console.log('❌ 运行时错误:');
  errors.forEach(e => console.log('   ' + e));
} else {
  console.log('✅ 无运行时错误');
}

await browser.close();
